asistente_ladm_col/logic/quality/logic_quality_rules.py

```diff
@@ -0,0 +1,494 @@
+                                                                   2: record['suma_fracciones'],
+                                                                   3: self.quality_rules_manager.get_error_message(QUALITY_RULE_ERROR_CODE_E400301),
+                                                                   4: QUALITY_RULE_ERROR_CODE_E400301})
+                new_features.append(new_feature)
+            error_layer.dataProvider().addFeatures(new_features)
+        else:
+            self.logger.error_msg(__name__, "Error executing query for rule {}: {}".format(rule.rule_name, records))
+
+        return self.return_message(db, new_features, rule.error_table_name, error_layer, False)
+
+    def check_parcels_with_invalid_department_code(self, db):
+        rule = self.quality_rules_manager.get_quality_rule(EnumQualityRule.Logic.DEPARTMENT_CODE_HAS_TWO_NUMERICAL_CHARACTERS)
+        res, records = self.get_ladm_queries(db.engine).get_parcels_with_invalid_department_code(db)
+        if res:
+            return self.basic_logic_validations(db, records, rule, QUALITY_RULE_ERROR_CODE_E400401)
+
+    def check_parcels_with_invalid_municipality_code(self, db):
+        rule = self.quality_rules_manager.get_quality_rule(EnumQualityRule.Logic.MUNICIPALITY_CODE_HAS_THREE_NUMERICAL_CHARACTERS)
+        res, records = self.get_ladm_queries(db.engine).get_parcels_with_invalid_municipality_code(db)
+        if res:
+            return self.basic_logic_validations(db, records, rule, QUALITY_RULE_ERROR_CODE_E400501)
+
+    def check_parcels_with_invalid_parcel_number(self, db):
+        rule = self.quality_rules_manager.get_quality_rule(EnumQualityRule.Logic.PARCEL_NUMBER_HAS_30_NUMERICAL_CHARACTERS)
+        res, records = self.get_ladm_queries(db.engine).get_parcels_with_invalid_parcel_number(db)
+        if res:
+            return self.basic_logic_validations(db, records, rule, QUALITY_RULE_ERROR_CODE_E400601)
+
+    def check_parcels_with_invalid_previous_parcel_number(self, db):
+        rule = self.quality_rules_manager.get_quality_rule(EnumQualityRule.Logic.PARCEL_NUMBER_BEFORE_HAS_20_NUMERICAL_CHARACTERS)
+        res, records = self.get_ladm_queries(db.engine).get_parcels_with_invalid_previous_parcel_number(db)
+        if res:
+            return self.basic_logic_validations(db, records, rule, QUALITY_RULE_ERROR_CODE_E400701)
+
+    def check_invalid_col_party_type_natural(self, db):
+        rule = self.quality_rules_manager.get_quality_rule(EnumQualityRule.Logic.COL_PARTY_NATURAL_TYPE)
+        error_layer = None
+        error_layer_exist = False
+
+        group = self.qgis_utils.get_error_layers_group()  # Check if error layer exist
+        layers = group.findLayers()  # Check if layer is loaded
+        for layer in layers:
+            if layer.name() == rule.error_table_name:
+                error_layer = layer.layer()
+                error_layer_exist = True
```

Comment:
Same here, please remove this check and always add a new error layer.

Reply:
Thank you, resolved in e9e311a